packages/fuel-gauge/src/abi/vitest.matcher.ts

```diff
@@ -0,0 +1,20 @@
+import { bn } from 'fuels';
+import type { BNInput } from 'fuels';
+
+export const toEqualBn = (_received: BNInput, _argument: BNInput) => {
+  const received = bn(_received);
+  const argument = bn(_argument);
+
+  const pass = received.eq(argument);
+
+  if (pass) {
+    return {
+      message: () => `Expected ${received.toString()} not to equal ${argument.toString()}`,
+      pass: true,
+    };
+  }
+  return {
+    message: () => `expected ${received.toString()} to equal ${argument.toString()}`,
+    pass: false,
+  };
+};
```

Comment:
🚀

Reply:
@nedsalk said this wasn't going to go down well with the team

Reply:
The annoying (ugly) part of adding these custom assertions is usually making TS aware of them.

Reply:
@arboleya are you happy with the Vitest matcher, or would you favour a standalone function?